mafContrib/updatedDitherStackers.py

```diff
@@ -0,0 +1,1196 @@
+
+# Some of the stackers here are modified versions of the stackers in lsst.sims.maf.stackers.ditherStackers.
+# These function similarly to ditherStackers,  with the major difference that dithers here are restricted to
+# within the hexagons used to tile the sky. Each of these hexagons is inscribed in a FOV, and having dithers
+# restricted to within these hexagons allows avoiding oversampling near the boundaries of the hexagonal tiles.
+ 
+# Also, the stackers here follow new naming scheme:  [Pattern]Dither[Field]Per[Timescale]. The absence of the
+# keyword 'Field' implies dither assignment to all fields.
+
+# Humna Awan: humna.awan@rutgers.edu
+# Date last modified: 04/09/16
+###############################################################################################################
+import numpy as np
+import matplotlib.pyplot as plt
+from lsst.sims.maf.stackers import BaseStacker
+from mafContrib.SeasonStacker_v2 import SeasonStacker_v2 as SeasonStacker
+import time
+
+__all__ = ['RandomDitherFieldPerVisitStacker',
+           'RepulsiveRandomDitherFieldPerVisitStacker',
+           'SpiralDitherFieldPerVisitStacker',
+           'SequentialHexDitherFieldPerVisitStacker',
+           'RandomDitherFieldPerNightStacker',
+           'RepulsiveRandomDitherFieldPerNightStacker',
+           'SpiralDitherFieldPerNightStacker',
+           'SequentialHexDitherFieldPerNightStacker',
+           'RandomDitherPerNightStacker',
+           'RepulsiveRandomDitherPerNightStacker',
+           'SpiralDitherPerNightStacker',
+           'SequentialHexDitherPerNightStacker',
+           'PentagonDitherFieldPerSeasonStacker',
+           'PentagonDiamondDitherFieldPerSeasonStacker',
+           'PentagonDitherPerSeasonStacker',
+           'PentagonDiamondDitherPerSeasonStacker',
+           'SpiralDitherPerSeasonStacker',
+           'FermatSpiralDitherFieldPerVisitStacker',
+           'FermatSpiralDitherFieldPerNightStacker',
+           'FermatSpiralDitherPerNightStacker']
+
+def wrapRADec(ra, dec):
+    """
+    Wrap RA and Dec values so RA between 0-2pi (using mod),
+      and Dec in +/- pi/2.
+    """
+    # Wrap dec.
+    low = np.where(dec < -np.pi/2.0)[0]
+    dec[low] = -1 * (np.pi + dec[low])
+    ra[low] = ra[low] - np.pi
+    high = np.where(dec > np.pi/2.0)[0]
+    dec[high] = np.pi - dec[high]
+    ra[high] = ra[high] - np.pi
+    # Wrap RA.
+    ra = ra % (2.0*np.pi)
+    return ra, dec
+
+def wrapRA(ra):
+    """
+    Wrap only RA values into 0-2pi (using mod).
+    """
+    ra = ra % (2.0*np.pi)
+    return ra
+
+def polygonCoords(nside,radius, rotationAngle):
+    """
+    Find the x,y coords of a polygon.
+    """
+    eachAngle= 2*np.pi/nside
+    xCoords= np.zeros(nside, float)
+    yCoords= np.zeros(nside, float)
+    for i in range(0,nside):
+        xCoords[i]= np.sin(eachAngle*i + rotationAngle)*radius
+        yCoords[i]= np.cos(eachAngle*i + rotationAngle)*radius
+        
+    return zip(xCoords,yCoords)
+
```

Comment:
These functions should be imported from core maf, rather than reimplemented here. This guards against changes in definition of ra/dec, and incorporates bugfixes that may arise there. These functions are pretty simple, so it's less likely such things would happen, but it's good practice not to duplicate code.